Applications/WebPortal/WebApp/library/source/pages/account_page/entitlements_page/entitlement_table.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+import * as Nexus from 'nexus';
+import * as Beam from 'beam';
+import {EntitlementsPageSizing} from './entitlements_page';
+
+interface Properties {
+  entitlementEntry: Nexus.EntitlementDatabase.Entry;
+  breakpoint: EntitlementsPageSizing.BreakPoint;
+}
+
+export class EntitlementTable extends React.Component<Properties, {}> {
+  constructor(properties: Properties) {
+    super(properties);
+  }
+
+  public render(): JSX.Element {
+    const containerWidth = (() => {
+      if(this.props.breakpoint === EntitlementsPageSizing.BreakPoint.SMALL) {
+        return EntitlementTable.STYLE.container.small;
+      } else {
+        return EntitlementTable.STYLE.container.notSmall;
+      }
+    })();
+    const compactHeader = (() => {
+      if(this.props.breakpoint === EntitlementsPageSizing.BreakPoint.SMALL) {
+       return EntitlementTable.STYLE.headerLabel;
+      } else {
+        return EntitlementTable.STYLE.hiddenHeader;
+      }
+    })();
+    const expandedHeader = (() => {
+      if(this.props.breakpoint === EntitlementsPageSizing.BreakPoint.SMALL) {
+        return EntitlementTable.STYLE.hiddenHeader;
+      } else {
+        return EntitlementTable.STYLE.headerLabel;
+      }
+    })();
+    const tableData = (() => {
+      const data = [];
+      for(const applicability of this.props.entitlementEntry.applicability) {
+        const dots = [];
+        const greenCircle =
+          (<td>
+            <img style={EntitlementTable.STYLE.circle}
+              src=
+                'resources/account_page/entitlements_page/icons/dot-green.svg'/>
+          </td>);
+        const greyCircle =
+          (<td>
+            <img style={EntitlementTable.STYLE.circle}
+              src=
+                'resources/account_page/entitlements_page/icons/dot-grey.svg'/>
+          </td>);
+        if(applicability[1].test(Nexus.MarketDataType.BBO_QUOTE)) {
+          dots.push(greenCircle);
+        } else {
+          dots.push(greyCircle);
+        }
+        if(applicability[1].test(Nexus.MarketDataType.MARKET_QUOTE)) {
+          dots.push(greenCircle);
+        } else {
+          dots.push(greyCircle);
+        }
+        if(applicability[1].test(Nexus.MarketDataType.BOOK_QUOTE)) {
+          dots.push(greenCircle);
+        } else {
+          dots.push(greyCircle);
+        }
+        if(applicability[1].test(Nexus.MarketDataType.TIME_AND_SALE)) {
+          dots.push(greenCircle);
+        } else {
+          dots.push(greyCircle);
+        }
+        if(applicability[1].test(Nexus.MarketDataType.ORDER_IMBALANCE)) {
+          dots.push(greenCircle);
+        } else {
+          dots.push(greyCircle);
+        }
+        data.push(
+          <tr style={EntitlementTable.STYLE.row}>
+            <td>{applicability[0].source.toString()}</td>{dots}
```

Comment:
This ```toString``` returns the market identifier code (MIC). This code is useful for computers but often not very helpful for humans. You need to display the common displayName. To get the displayName refer to the Fogbugz case assigned to you that goes through the lengthy process involving looking up the displayName from a MarketDatabase. Take that process and wrap it into function and use it.